saves your day if you get unexplainable errors from rpm install because /vservers/test has bad permissions:

# la -d /vservers/test/
drwx------  23 root root 4096 Jun 16 02:01 /vservers/test/

# vrpm test -- -Uhv masqmail-0.2.20-5.amd64.rpm
Preparing...                ########################################### [100%]
rpm-fake-resolver: getpwnam(): Invalid argument
   1:masqmail               ########################################### [100%]
warning: user mail does not exist - using root
warning: user mail does not exist - using root
warning: user mail does not exist - using root
warning: user mail does not exist - using root
#

real fix would be 'chmod 755 /vservers/test' if you just knew that is the problem!

--- util-vserver-0.30.210/src/rpm-fake-resolver.c~	2006-06-16 03:22:36.000000000 +0300
+++ util-vserver-0.30.210/src/rpm-fake-resolver.c	2006-06-16 03:22:40.000000000 +0300
@@ -301,10 +301,16 @@
   Echdir("/");
 
   activateContext(args.ctx, args.in_ctx, args.caps, args.flags);
   Esetgroups(0, &args.gid);
   Esetgid(args.gid);
   Esetuid(args.uid);
 
+  if (access("/", X_OK) == -1) {
+    WRITE_MSG(2,
+    "***  root directory is not accessible for specified uid. rpm-fake-resolver will most likely fail.\n"
+    );
+  }
+
   if (args.do_fork) daemonize(&args, pid_fd);
   if (pid_fd!=-1)   close(pid_fd);
   run();
